Package.swift

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+//
+// swift-tools-version: 5.7
+//   The swift-tools-version declares the minimum version of Swift required to build this package.
+//
+// A user of the Swift Package Manager (SPM) package will consume this file directly from the ORT github repository.
+// For context, the end user's config will look something like:
+//
+//     dependencies: [
+//       .package(url: "https://github.com/microsoft/onnxruntime", branch: "rel-1.14.0"),
+//       ...
+//     ],
+//
+// NOTE: The direct consumption creates a somewhat complicated setup to 'release' a new version of the ORT SPM package.
```

Comment:
Yes, very much doable! I love this idea :)

Reply:
I think there are pros and cons to each. 

The python script is possibly more easily understood/used/debugged for the average ORT developer, and it provides a simple way to ensure the checksum is correct when updating the config to point to a new release. 

If we do it via swift and environment variables that seems like it might be a little harder to make/test changes as it requires something that can run swift. Still requires something that a release manager can easily run to calculate the checksum and update this file correctly as part of the release process though. 

Either way it doesn't look like we can avoid the multi-step release process, so I'd err on the side of what's easiest for us to test and maintain which I _think_ slightly favours the python script but open to more discussion about what the best approach is.

Reply:
Feels like we're getting into a lot of complexity just to avoid updating Package.swift post-release and cherry-picking that change to the release branch, and if we did that, the version of the file on `main` is always pointing at the latest release and the tag is as expected and doesn't need any special explanation.